### PR TITLE
support concurrent build in maven plugin

### PR DESCRIPTION
**Pull request by piotrwielgolaski-tomtom**
## Impact

- [x] Bug fix (non-breaking change which fixes expected existing functionality)
- [ ] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 

## Description

This is remaining part of https://github.com/liquibase/liquibase/issues/3446

To make liquid base work with maven concurrent build (-T 1C)

It uses scope manager, but it need to be set in static so before any plugins action kick in.
Also it should not inherit any scope, so each plugin execution can start on clean scope isolated from another.

### CODE REVIEW

**Comment by piotrwielgolaski-tomtom** [comment]
@filipelautert I noticed issue with test, fixed it now

**Comment by piotrwielgolaski-tomtom** [comment]
@filipelautert any chance to get this under review?

**Comment by trico-tillster** [comment]
@piotrwielgolaski-tomtom I still see this issue with latest version 4.23 using -T2. Will your PR fix this? Thank you
`Error:  Failed to execute goal org.liquibase:liquibase-maven-plugin:4.23.0:update (liquibase) on project ordering-db-schema: Cannot end scope ztupxdjtqp when currently at scope root`

**Comment by piotrwielgolaski-tomtom** [comment]
> @piotrwielgolaski-tomtom I still see this issue with latest version 4.23 using -T2. Will your PR fix this? Thank you

 Biased on my testing, yes it should, but keep in mind that I focused on offline database use case. 
 anyway for sure it will bring us closer for all cases.

**Comment by trico-tillster** [comment]
@piotrwielgolaski-tomtom I've patched liquibase with this fix and I can confirm that the scope issue regarding maven concurrency that I was observing in 4.23.0 is fixed. Thank you
@filipelautert Thank you for picking this up and move it forward.

File: liquibase-standard/src/main/java/liquibase/Scope.java
Comment (filipelautert) [inline liquibase-standard/src/main/java/liquibase/Scope.java:116]: Besides the "default value" is removed here, the getCurrentScope later on has similar logic to make sure that we don't have a null scope.

File: liquibase-standard/src/main/java/liquibase/command/core/helpers/AbstractOutputWriterCommandStep.java
Comment (filipelautert) [inline liquibase-standard/src/main/java/liquibase/command/core/helpers/AbstractOutputWriterCommandStep.java:20]: Thanks, this static was incorrect .

**Review by filipelautert — APPROVED** [review]

File: liquibase-standard/src/main/java/liquibase/Scope.java
Comment (piotrwielgolaski-tomtom) [reply liquibase-standard/src/main/java/liquibase/Scope.java:116]: It is important to not inject common scope for scope manager, as all child will inherit from it, which you don't want if scope manager should have isolated scope.

**Comment by piotrwielgolaski-tomtom** [comment]
@rberezen it looks it is waiting for your approval, please find time to take a look at.

**Comment by filipelautert** [comment]
Hello @piotrwielgolaski-tomtom - our build process is going through some refactoring and it is blocking @rberezen . Once it is done/fixed he will be able to finish the review here... Thanks for your patience!

**Review by rberezen — APPROVED** [review]

**Pull request merged**
